### PR TITLE
Feature/CP-467

### PR DESCRIPTION
**Pull request by camsjams**
Use Lambda function in API gateway to reduce reliance on SF creds

Also see [cpma-tax-and-payment PR 40](https://github.com/MakerStudios/cpma-tax-and-payment/pull/40)

### CODE REVIEW

File: lib/Connection.js [hunk 1]
```diff
@@ -63,42 +60,73 @@ class Connection {
...
+		return this.getSignedHeaders(body)
+			.then((headers) => {
+				return fetch(`${PROTOCOL}${this.config.host}${this.config.path}`, {
+					method: 'POST',
+					headers,
+					body
+				});
+			})
+			.then((response) => {
+				if (response.status >= 400) {
```
Comment (cleathers) [inline lib/Connection.js]: Should this be `!response.ok`?

File: lib/Connection.js [hunk 2]
```diff
@@ -121,13 +149,14 @@ class Connection {
...
-		}).then((response) => {
-			if ((response || {}).status === 200) {
-				return true;
-			}
-
-			throw new Error(ACCESS_TOKEN_INVALID);
-		});
+		})
+			.then((response) => {
+				if ((response || {}).status === 200) {
```
Comment (cleathers) [inline lib/Connection.js]: Should this be `response.ok`?

**Comment by cleathers** [comment]
Looks good. If we could use `response.ok` in those couple of place that would be cool, but if it's not available then I'd say lets merge it

File: lib/Connection.js (hunk 1 above)
Comment (camsjams) [inline lib/Connection.js]: it can and will

**Comment by camsjams** [comment]
@cleathers I'm ok with the `response.ok` :+1:

**Comment by camsjams** [comment]
updated

**Pull request merged**